### DISCUSSION TITLE
Secp256k1 curves in JWK format with browsers

### DISCUSSION THREAD

**Discussion by 0xCaso** [panva/jose#439] [Q&A] [answered]
Is there a way I can do something like this using a browser as a runtime (specifically I'm using ReactJS)
```
const pvt = {
    "kty": "EC",
    "d": "1rkwp18RoEPcCqU8bHGUnLKe_8Yna5nRLCgG827WbpM",
    "use": "sig",
    "crv": "secp256k1",
    "kid": "5c1e35deff5746ce84c8791b94079cab",
    "x": "Jgqna25VztAel1PhmbnZXAbji2MIbgGqh4fS3sD7BME",
    "y": "Zs_20cwed2-xg8Mc2G4OwEAa1XtGo9fLgVkKxFzDIJM",
    "alg": "ES256K"
}
const privateKey = await jose.importJWK(pvt, "ES256K");
```
I'd have to sign a message using a private key like this and I have no idea how to do it.
Maybe using Node.js as a backend would fix the problem?

**Comment by panva** [comment, marked as answer]
As per [Browser Support](https://github.com/panva/jose/issues/263), the Web Cryptography API used in a browser does not support the `secp256k1` EC curve.

Therefore this won't work in any Web Cryptography API backed runtime, only Node.js.
